main.py

```diff
@@ -69,7 +70,10 @@ def main(username, scope, clientID, clientSecret, redirectURI, path):
         except TypeError:
             pass
         
-        time.sleep(1)
+        time.sleep((spotify.current_user_playing_track['duration_ms']/1000) - SubtractDelay) #Stop us from getting rate limited from spotify API
+        #Instead of checking every second for an ad. We simply get the total length of the song and wait for that ammount of time. 
+        #Now spotify returns miliseconds for song length which is weird but this can easily be fixed with some basic math.
+        #We get the song duration and divide it by 1000 and we get our time in seconds
```

Comment:
These are unnecessary, the comment on line 73 is enough (it explains WHY, the code already explains HOW and the names are explicit enough).